django_socio_grpc/proto_serializers.py

```diff
@@ -65,23 +68,65 @@ def populate_dict_with_none_if_not_required(self, data_dict, message=None):
                 code="missing_partial_message_attribute",
             )
 
+        is_update_process = bool(data_dict.get(self.Meta.model._meta.pk.name, ""))
         for field in self.fields.values():
-            # INFO - AM - 04/01/2024 - If we are in a partial serializer we need to only have field specified in PARTIAL_UPDATE_FIELD_NAME attribute. Meaning bot deleting field that should not be here and not adding None to allow_null field that are not specified
-            if (
-                message
-                and self.partial
-                and hasattr(message, PARTIAL_UPDATE_FIELD_NAME)
-                and field.field_name not in getattr(message, PARTIAL_UPDATE_FIELD_NAME)
+            # INFO - AM - 04/01/2024 - If we are in a partial serializer we only need to have field specified in PARTIAL_UPDATE_FIELD_NAME attribute in the data. Meaning deleting fields that should not be here and not adding None to allow_null field that are not specified
+            if self.partial and field.field_name not in data_dict.get(
+                PARTIAL_UPDATE_FIELD_NAME, {}
             ):
                 if field.field_name in data_dict:
                     del data_dict[field.field_name]
                 continue
             # INFO - AM - 04/01/2024 - if field already existing in the data_dict we do not need to do something else
             if field.field_name in data_dict:
                 continue
-            # INFO - AM - 04/01/2024 - Adding default None value only for optional field that are required and allowing null or having a default value
-            if field.allow_null or field.default in [None, empty] and field.required is True:
+
+            if self.partial and field.field_name in data_dict.get(
+                PARTIAL_UPDATE_FIELD_NAME, {}
+            ):
+                if field.allow_null:
+                    data_dict[field.field_name] = None
+                    continue
+                if field.default not in [None, empty]:
+                    # TODO - AM - 12/01/2024 - is field.default a possible callable here ?
+                    data_dict[field.field_name] = field.default
+                    continue
+
+                data_dict[field.field_name] = message.DESCRIPTOR.fields_by_name[
```

Comment:
this should throw imo, that means you explicitely put null into a non nullable field

Reply:
may worth the shot to delete it and see which test fail